### ISSUE TITLE
MAMC implement

### ISSUE THREAD

**Issue by zhaoxin111** [xcnkx/fine_grained_classification#1]
Hi，I am going to reimplement the paper：Multi-Attention Multi-Class Constraint for
Fine-grained Image Recognition recently.

When I implement the MAMC module，I don't know how to implement it with N pair loss? Do you have the code to implement MAMC?

**Comment by xcnkx** [comment]
Hi, @zhaoxin111. 
But I just implemented it without the MAMC loss function so I don't have the implemented code. :(

**Comment by zhaoxin111** [comment]
> Hi, @zhaoxin111.
> But I just implemented it without the MAMC loss function so I don't have the implemented code. :(
It doesn't matter, your work is very good. I am trying to reimplement MAMC myself. :)

**Comment by fxle** [comment]
Hello,Have you finished the MAMC module?I'm interested in it, too. @zhaoxin111

**Comment by zhaoxin111** [comment]
> Hello,Have you finished the MAMC module?I'm interested in it, too. @zhaoxin111

not yet:(

**Comment by RQuispeC** [comment]
Hello fellows, I implemented n-pairs loss as part of an idea I'm testing.

https://github.com/RQuispeC/npairs_loss_MAMC

Let me know if you find any bug or something weird. I'm testing it in another task and everything seems smooth.

**Comment by zahngyoujia** [comment]
Hello,Have you finished the MAMC module?Could you share your code?

**Comment by zahngyoujia** [comment]
Hello,Have you finished the MAMC module?Could you share your code? @zhaoxin111

**Comment by xcnkx** [comment]
>Hello,Have you finished the MAMC module?Could you share your code? @zhaoxin111 

Hello @zahngyoujia , sorry man but I have no plans to implement the MAMC module now. One reason is that without this module, only with the multi-attention module, it was enough for my project.

But I found one guy (https://github.com/RQuispeC/npairs_loss_MAMC) that implemented the MAMC module in Pytorch. You can use it as a reference to implement it in Tensorflow or Keras.
